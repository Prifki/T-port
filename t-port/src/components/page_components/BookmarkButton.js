import React, { Component } from 'react';

class BookmarkButton extends Component {
  render() {
    return (
        <div className="card__button--bookmark">
            <i className="material-icons">bookmark_border</i>
        </div>
    );
  }
}

export default BookmarkButton;
