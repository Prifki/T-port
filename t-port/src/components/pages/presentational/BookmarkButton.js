import React from 'react';

function BookmarkButton(props) {
    return (
        <div className="card__button--bookmark" onClick={props.onClick}>
            <i className="material-icons">{props.type}</i>
        </div>
    );
}

export default BookmarkButton;
