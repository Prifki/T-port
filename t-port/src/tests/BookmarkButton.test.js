import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import BookmarkButton from './../components/pages/presentational/BookmarkButton';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('BookmarkButton', () => {
    it('should render correctly', () => {
      const output = shallow(
        <BookmarkButton type="star_border" />
      );
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});

