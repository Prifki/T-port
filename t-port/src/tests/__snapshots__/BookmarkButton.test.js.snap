// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkButton should render correctly 1`] = `
<div
  className="card__button--bookmark"
>
  <i
    className="pictorams"
  >
    star_border
  </i>
</div>
`;

exports[`LoginMenu should render correctly 1`] = `
<div
  className="card__button--bookmark"
>
  <i
    className="pictorams"
  >
    star_border
  </i>
</div>
`;
